### DISCUSSION TITLE
Actividad 3 - Grupo 3

### DISCUSSION THREAD

**Discussion by renatacatelli** [frlp-utn-ingsoft/catedra#128] [Show and tell]
https://github.com/BogadoLucasDemian/vetsoft

Buenas, realizo la entrega de la Actividad 3 del Grupo 3. Quería aclarar que cuando yo estaba trabajando en el feature a testear mío (validar que el precio de un producto sea mayor a cero) cuando trabajé aceptando y resolviendo conflictos, se fueron agregando los archivos al stage area y cuando hice el commit, se hicieron todos en uno solo: https://github.com/BogadoLucasDemian/vetsoft/commit/91669e397b665a984227c895b04f011d1b363a0c

No me había pasado antes y una vez que resolví los conflictos, tendría que haber hecho los commits por tests para que quede todo ordenado.

**Comment by ivanduva** [comment]
Buenas!
Si, vimos ese problema de procedimiento. Además de eso también en el mismo PR se agregan cambios relativos a docker que tampoco está bueno porque no tienen nada que ver con la funcionalidad que se quería agregar. También vemos nombres de commit duplicados, lo cual es confuso. Por último, hay indicios de una mala review en ese PR porque vemos que se encontraron errores en la edición que otra persona solucionó en un PR aparte, lo cual tendría que haberse solucionado en el mismo PR por la persona encargada si la review hubiera sido buena.

Además de esto, vemos otros problemas que deberían solucionar para aprobar la actividad:

- Al editar el producto, borra el precio y te pide que ingreses algo. Debería quedar el precio que ya existe. También estaría bueno agregar tests de unidad e integración relativos a cuando se intenta agregar un precio con error.
- En cuanto a la especialidad del veterinario, faltan tests específicos de unidad e integración con respecto a ese atributo. Los tests están bien pero son más generales sobre la clase Vet. Cuando se soluciona un issue, se debería agregar tests relativos a la funcionalidad que describía ese issue.

Cada problema debe solucionarlo la persona que se encargó de la funcionalidad.

En cuanto al trabajo en las reviews de los PR, sumando a lo que les comentaba al principio, vemos otros problemas. Por ejemplo, en varios aprobaron el PR y dejaron comentarios atinados "para el futuro" cuando la idea es dar un feedback que permita refactorizar el código o minimamente discutirlo si no se está de acuerdo. El procedimiento que creemos más apropiado lo vemos acá: https://github.com/BogadoLucasDemian/vetsoft/pull/10 , donde se hizo un comentario y se actuó en consecuencia, si bien por ahí faltaría marcar como resuelto el comentario o responderlo.
Todo esto deberían tenerlo en cuenta para mejorar su proceso en próximas entregas.

Como comentario menor, vemos un comportamiento raro en algunos casos al validar los forms: se borra el campo que tiene error. En general  se suele marcar el error sin borrar lo que el usuario ingresó.

El resto está bien, la app funciona, los tests pasan y además hicieron bastantes y lo valoramos.
Saludos!

**Reply by RoTantos** [reply]
Buenas! Ya solucionamos los problemas mencionados. 
Saludos!

**Comment by ivanduva** [comment]
Buenas
Perfecto, queda aprobado.
Una observación para las próximas entregas: si van a hacer comentarios en los PR, traten que sean relevantes. Comentarios como "Los tests funcionan" o "Funciona bien" no aportan nada a una review. Para eso esta el approve que también lo están utilizando.

Saludos!
